SPECK/src/main/java/SPEck.java

```diff
@@ -303,85 +383,104 @@ public static String[] parallelGenDataset(int i, int T, double theta, String str
-     * @param   parallelization the number of cores of the machine. It is the number of random datasets
+     * @param   numCores the number of cores of the machine. It is the number of random datasets
      *                          generated at the same time in the Monte Carlo estimation. Each core of
-     *                          the machine computes the p-values on T/parallelization random datasets.
+     *                          the machine computes the p-values on T/numCores random datasets.
      * @param theta             the minimum frequency threshold used to mine the dataset
      * @param fwer              the maximum family-wise error rate (FWER) threshold used to mine the dataset
      */
-    void execute(int P, int T, int parallelization, double theta, double fwer) throws IOException, ClassNotFoundException, InterruptedException {
-        String file = fileIn.split("\\.")[0];
+    void execute(int P,
+                 int T, 
+                 int numCores, 
+                 double theta, 
+                 double fwer
+    ) throws IOException,
+             ClassNotFoundException, 
+             InterruptedException 
+    {
+        String file = obsDatasetPath.split("\\.")[0];
         String fileRandom = file + "_random.txt";
         String fileMinned = file + "_mined.txt";
         // loads the input dataset in the data structures provided in input
-        loadDataset(fileIn,positionOrigin,datasetOrigin);
+        loadDataset(obsDatasetPath,positionOrigin,datasetOrigin);
         // fixes the seed for the first random generator
         Random r = new Random();
         System.err.println("Generating t datasets...");
-        generateTDatasets(T, theta, parallelization);
+        generateTDatasets(T, theta, numCores);
+
         double[] minPvalue = new double[P];
         // for all the P datasets of the WY method
         System.err.println("Calculating p values for corrected threshold in spark...");
         for(int j=0;j<P;j++) {
-            datasetRandom = Utils.generateDataset(datasetOrigin, positionOrigin, strategy, r);
+            Utils.Dataset datasetRandom = Utils.generateDataset(datasetOrigin, positionOrigin, strategy, r);
             // writes the actual random dataset generated
             writeDataset(fileRandom,datasetRandom);
             // mines the actual random dataset generated
             mining(fileRandom, fileMinned, theta);
             // computes in parallel the p-values of the actual random dataset
-            int[] pValueInt = computePValues(T,parallelization,fileMinned);
+            int[] pValueInt = computePValues(T,numCores,fileMinned);
             // finds the minimum of the p-values computed from the actual random dataset
             int min = Integer.MAX_VALUE;
             for (int i : pValueInt) {
                 if (min > i) min = i;
             }
             minPvalue[j] = (1 + min) / (T * 1. + 1);
```

Comment:
Add a comment above this line: "It's only here that the number of T datasets with results as or more extreme as the one from the random dataset is normalized into a p-value".